cairo/tests/ethereum/cancun/vm/instructions/test_environment.py

```diff
@@ -107,6 +115,37 @@ def codecopy_strategy(draw):
     return evm
 
 
+@composite
+def calldatacopy_strategy(draw):
+    """Generate test cases for the calldatacopy instruction.
+
+    This strategy generates an EVM instance and the required parameters for calldatacopy.
+    - 8/10 chance: pushes all parameters onto the stack to test normal operation
+    - 2/10 chance: use stack already populated with values, mostly to test error cases
+    """
+    evm = draw(
+        EvmBuilder()
+        .with_stack()
+        .with_gas_left()
+        .with_memory()
+        .with_message(strategy=message_empty_except_calldata)
+        .build()
+    )
+
+    memory_start_index = draw(memory_lite_start_position)
+    data_start_index = draw(uint256)
+    size = draw(uint256)
+
+    # 80% chance to push valid values onto stack
+    should_push = draw(integers(0, 99)) < 80
+    if should_push:
+        push(evm.stack, U256(size))
```

Comment:
why would these values be 'valid'?

Reply:
done with [c87f4b9](https://github.com/kkrt-labs/keth/pull/504/commits/c87f4b90b2bd309e08532f44a338b2facb684021)